### ISSUE TITLE
Idea: run-parts (or similar) against /rw/config/rc.local.d/

### ISSUE THREAD

**Issue by rysiekpl** [QubesOS/qubes-issues#5715]
**The problem you're addressing (if any)**
The more advanced my use of QubesOS becomes, the more often I need to add random code to `/rw/config/rc.local`. Even though I usually do that via Salt, it becomes unmanageable pretty fast, with `rc.local` becoming a spaghetti code mess.

**Describe the solution you'd like**

Using an approach similar to Debian's `run-parts` as used in `/etc/rc.local` would make it possible to add discrete scripts to `/rw/config/rc.local.d` to be executed on vm boot.

The relevant line from Debian's `/etc/rc.local`:

```
[ -d /etc/boot.d ] && run-parts /etc/boot.d
```
Obviously a more generic solution (working on Fedora-based images) would be in order.

**Where is the value to a user, and who might that user be?**

Adding and, more importantly, managing changes to code run on vm startup would be easier and more transparent. This would bring most benefit to users using the Salt management stack (since managing files is easier than managing specific lines in a single file), but any user who needs to add startup code on their vms would benefit from additional clarity.

**Describe alternatives you've considered**

Obviously the current alternative is to just add all the relevant code directly to `/rw/config/rc.local`, but that doesn't scale.

**Additional context**
This might be especially useful in context of tools like `split-ssh`, which require code to run on vm startup.

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**

https://www.qubes-os.org/doc/salt/

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**

None that I could find.

**Comment by marmarek** [comment]
The obvious thing you can do is to add `run-parts /rw/config/rc.local.d` to your `/rw/config/rc.local`. You can also use [bind-dirs](https://www.qubes-os.org/doc/bind-dirs/) to put files into `/etc/boot.d`.

**Comment by rysiekpl** [comment]
Yeah, I will be playing with both approaches tomorrow. But it seems to me an issue generic enough to have a ticket; I would bet a lot of Qubes users rolled out some bespoke solutions to this.

Feels like it would make sense to have a standard solution in Qubes, especially since:
1. it would basically be a one-line addition to `/rw/config/rc.local`
2. I don't really see any serious security implications of such a change

Having it as a standard Qubes thing, working reliably in all official templates, would mean authors of solutions like [`split-ssh`](https://github.com/henn/qubes-app-split-ssh) have a standard, supported way of adding scripts that run on VM boot.
